File: src/backend/MyAIRunningMate/MyAIRunningMate.Application/Models/ViewObjects/TrainingPlanView.cs
namespace MyAIRunningMate.Application.Models.ViewObjects;

public class TrainingPlanView
{
    public string Title { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; } 
    public string Description { get; set; }
    public IEnumerable<TrainingPlanEventView> TrainingPlanEvents { get; set; }
}

File: src/frontend/MyAIRunningMate-UI/src/types/views/trainingPlanView.ts
export interface TrainingPlanEventView {
  eventDate: string;
  exerciseType: string;
  exerciseSubtype: string;
  description: string;
  distanceMetres: number;
}

export interface TrainingPlanView {
  title: string;
  startDate: string;
  endDate: string;
  description: string;
  trainingPlanEvents: TrainingPlanEventView[];
}
